plugins/webp-uploads/hooks.php

```diff
@@ -719,7 +719,13 @@ function webp_uploads_get_image_sizes_additional_mime_type_support(): array {
 	);
 
 	foreach ( $additional_sizes as $size => $size_details ) {
-		$allowed_sizes[ $size ] = ! empty( $size_details['provide_additional_mime_types'] );
+		if ( isset( $size_details['provide_additional_mime_types'] ) ) {
+			// Give priority to the 'provide_additional_mime_types' property.
+			$allowed_sizes[ $size ] = (bool) $size_details['provide_additional_mime_types'];
+		} else {
+			// If 'provide_additional_mime_types' is not set, use the fallback all sizes setting.
+			$allowed_sizes[ $size ] = webp_uploads_is_fallback_all_sizes_enabled();
+		}
```

Comment:
Rather than doing this here, I think a more elegant solution would be to decouple the UI control and option from the main logic, by using the `webp_uploads_image_sizes_with_additional_mime_type_support` filter below. So I think this would be better separately implemented.

It would also allow the code to be a bit more efficient: We should only need to call `webp_uploads_is_fallback_all_sizes_enabled()` once, not for every size available. If `false` we can immediately return without changing anything. If `true`, we go through the array of sizes and set them to `true`.

I think it's okay not to check the `provide_additional_mime_types` value directly to see if it's `false`. It's primary purpose it to _enable_ (not disable) this behavior anyway, and since the setting explicitly says "all", it's fine to do it for all sizes, regardless of what other configuration may be present.